### PR TITLE
remove property type and save properties on process type

### PR DESCRIPTION
**Pull request by Kathrin-Huber**
Writes properties in ProcessType instead of indexing them separately which highly enhances the indexing-all speed.

This PR disables extendedSearch by Property, which will be fixed in a follow-up PR.

### CODE REVIEW

File: Kitodo/src/main/java/org/kitodo/production/services/data/PropertyService.java [hunk 1]
```diff
@@ -156,8 +115,8 @@ private QueryBuilder getQueryForType(String type) {
      *            of the searched property
      * @return list of JSON objects with properties
      */
-    public List<Map<String,Object>> findByTitle(String title, String type, boolean contains) throws DataException {
-        return findProperty(PropertyTypeField.TITLE.getKey(), title, type, contains);
+    public List<Property> findByTitle(String title, String type, boolean contains) {
+        return getByQuery("from Property as property where property.title='" + title + "'");
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/production/services/data/PropertyService.java:119]: Please use `getByQuery` with second parameter with list of to map values. If not this is introducing a possible SQL injection.

File: Kitodo/src/main/java/org/kitodo/production/services/data/PropertyService.java [hunk 2]
```diff
@@ -171,17 +130,8 @@ private QueryBuilder getQueryForType(String type) {
...
-
-    private List<Map<String,Object>> findProperty(String key, String value, String type, boolean contains) throws DataException {
-        BoolQueryBuilder query = new BoolQueryBuilder();
-        query.must(createSimpleQuery(key, value, contains, Operator.AND));
-        if (Objects.nonNull(type)) {
-            query.must(createSimpleQuery(PropertyTypeField.TYPE.getKey(), type, true));
-        }
-        return findDocuments(query);
+    List<Property> findByValue(String value, String type, boolean contains) {
+        return getByQuery("from Property as property where property.value='" + value + "'");
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/production/services/data/PropertyService.java:134]: Please use `getByQuery` with second parameter with list of to map values. If not this is introducing a possible SQL injection.

File: Kitodo/src/main/java/org/kitodo/production/services/data/PropertyService.java [hunk 3]
```diff
@@ -198,25 +148,8 @@ private QueryBuilder getQueryForType(String type) {
...
-    @Override
-    public PropertyDTO convertJSONObjectToDTO(Map<String, Object> jsonObject, boolean related) throws DataException {
-        PropertyDTO propertyDTO = new PropertyDTO();
-        propertyDTO.setId(getIdFromJSONObject(jsonObject));
-        propertyDTO.setTitle(PropertyTypeField.TITLE.getStringValue(jsonObject));
-        propertyDTO.setValue(PropertyTypeField.VALUE.getStringValue(jsonObject));
-        propertyDTO.setCreationDate(PropertyTypeField.CREATION_DATE.getStringValue(jsonObject));
-        return propertyDTO;
+    List<Property> findByTitleAndValue(String title, String value, String type, boolean contains) {
+        return getByQuery("from Property as property where property.title='" + title + "' and property.value='" + value + "'");
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/production/services/data/PropertyService.java:152]: Please use `getByQuery` with second parameter with list of to map values. If not this is introducing a possible SQL injection.

File: Kitodo/src/main/java/org/kitodo/production/services/data/PropertyService.java [hunk 4]
```diff
@@ -133,16 +92,17 @@ public Long countResults(Map filters) throws DataException {
...
         return findDistinctTitles("process");
     }
 
-    private List<String> findDistinctTitles(String type) throws DataException, DAOException {
-        return findDistinctValues(getQueryForType(type), "title.keyword", true, countDatabaseRows());
-    }
-
-    private QueryBuilder getQueryForType(String type) {
-        return createSimpleQuery(PropertyTypeField.TYPE.getKey(), type, true, Operator.AND);
+    private List<String> findDistinctTitles(String type) {
```
Comment (solth) [inline Kitodo/src/main/java/org/kitodo/production/services/data/PropertyService.java:99]: Parameter `type` is never used.

File: Kitodo/src/main/java/org/kitodo/production/services/data/PropertyService.java [hunk 5]
```diff
@@ -156,8 +116,10 @@ private QueryBuilder getQueryForType(String type) {
      *            of the searched property
      * @return list of JSON objects with properties
      */
-    public List<Map<String,Object>> findByTitle(String title, String type, boolean contains) throws DataException {
-        return findProperty(PropertyTypeField.TITLE.getKey(), title, type, contains);
+    public List<Property> findByTitle(String title, String type, boolean contains) {
```
Comment (solth) [inline Kitodo/src/main/java/org/kitodo/production/services/data/PropertyService.java:119]: Parameters `type` and `contains` are never used.

File: Kitodo/src/main/java/org/kitodo/production/services/data/PropertyService.java [hunk 6]
```diff
@@ -171,17 +133,10 @@ private QueryBuilder getQueryForType(String type) {
...
-    }
-
-    private List<Map<String,Object>> findProperty(String key, String value, String type, boolean contains) throws DataException {
-        BoolQueryBuilder query = new BoolQueryBuilder();
-        query.must(createSimpleQuery(key, value, contains, Operator.AND));
-        if (Objects.nonNull(type)) {
-            query.must(createSimpleQuery(PropertyTypeField.TYPE.getKey(), type, true));
-        }
-        return findDocuments(query);
+    List<Property> findByValue(String value, String type, boolean contains) {
```
Comment (solth) [inline Kitodo/src/main/java/org/kitodo/production/services/data/PropertyService.java:136]: See above.

File: Kitodo/src/main/java/org/kitodo/production/services/data/PropertyService.java [hunk 7]
```diff
@@ -198,25 +153,11 @@ private QueryBuilder getQueryForType(String type) {
...
-
-    @Override
-    public PropertyDTO convertJSONObjectToDTO(Map<String, Object> jsonObject, boolean related) throws DataException {
-        PropertyDTO propertyDTO = new PropertyDTO();
-        propertyDTO.setId(getIdFromJSONObject(jsonObject));
-        propertyDTO.setTitle(PropertyTypeField.TITLE.getStringValue(jsonObject));
-        propertyDTO.setValue(PropertyTypeField.VALUE.getStringValue(jsonObject));
-        propertyDTO.setCreationDate(PropertyTypeField.CREATION_DATE.getStringValue(jsonObject));
-        return propertyDTO;
+    List<Property> findByTitleAndValue(String title, String value, String type, boolean contains) {
```
Comment (solth) [inline Kitodo/src/main/java/org/kitodo/production/services/data/PropertyService.java:156]: See above.

**Pull request merged**